test/studies/ml/lib/Chai.chpl

```diff
@@ -366,23 +361,23 @@ module Chai {
             return grad;
         }
 
-        proc backwardBatch(deltas: [] Tensor(3), inputs: [] Tensor(3)): [] Tensor(3) {
+        proc ref backwardBatch(deltas: [] Tensor(3), inputs: [] Tensor(3)): [] Tensor(3) {
             var newDeltas: [deltas.domain] Tensor(3);
-            forall (delta,inputs,newDelta) in zip(deltas,inputs,newDeltas) {
+            forall (delta,inputs,newDelta) in zip(deltas,inputs,newDeltas) with (ref this) {
```

Comment:
This is interesting, I asked whether we should require this in https://github.com/chapel-lang/chapel/issues/23229#issuecomment-1710882522

Reply:
This seems loosely related to https://github.com/chapel-lang/chapel/issues/19211 and particularly its [most recent comment](https://github.com/chapel-lang/chapel/issues/19211#issuecomment-1236028838).